dev_tools/utils/nxdl_utils.py

```diff
@@ -127,23 +127,21 @@ def get_nx_namefit(hdf_name, name, name_any=False):
     """
     if name == hdf_name:
         return len(name) * 2
+    if re.search(r"^[0-9]", hdf_name) is not None:
+        # Don't match anything if the name starts with a number
+        return -1
 
     uppercase_parts = re.findall("[A-Z]+(?:_[A-Z]+)*", name)
 
+    regex_name = name
     for up in uppercase_parts:
-        name = name.replace(up, r"([a-zA-Z0-9_]+)")
+        regex_name = regex_name.replace(up, r"[a-z0-9_]+")
 
-    name_match = re.search(rf"^{name}$", hdf_name)
+    name_match = re.search(rf"^{regex_name}$", hdf_name)
```

Comment:
Currently NeXus allows periods, too, in the middle: ^[a-zA-Z0-9_]([a-zA-Z0-9_.]*[a-zA-Z0-9_])?$  Hence, if 'up' is not the beginning or end section of the name, we can be more relaxed.  
Note that this function is used to check for inheritance inside nxdl files, too, so we need this flexibility (of also allowing uppercase letters). Checking the naming convention again, I have found that it is only a recommendation to use only lower case letters, but in fact uppercase letters are also tolerated officially (with the note that not necessarily all applications will support them).

Reply:
I followed the regex from https://fairmat-nfdi.github.io/nexus_definitions/datarules.html#validitemname.
This means we match for `[a-zA-Z0-9_.]+` and don't allow `.` at the beginning or the end of the hdf_name (in that case we return -1 for anything we got), except if it's an exact match.